### PR TITLE
"Quick" javadoc & slight refactoring 

### PR DESCRIPTION
**Pull request by syntaxi**
Adds javadoc to every single class that needed it and touches it up in places.
Also a few small method refactoring and two large name changes

### CODE REVIEW

File: assets/prefabs/blocks/effectors/EffectorOne.prefab [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "parent": "GooeyDefence:PlainBlock",
-  "FireEffect": {
+  "FireEffector": {
     "drain": 5
   }
 }
```
Comment (vampcat) [inline assets/prefabs/blocks/effectors/EffectorOne.prefab:6]: Newline at the end of file :)

File: src/main/java/org/terasology/gooeyDefence/DefenceField.java [hunk 2]
```diff
@@ -213,4 +214,21 @@ public static boolean isFieldActivated() {
     public static void setFieldActivated() {
         DefenceField.fieldActivated = true;
     }
+
+    /**
+     * Helper method for getting a component given one of it's superclasses
```
Comment (vampcat) [inline src/main/java/org/terasology/gooeyDefence/DefenceField.java]: it's -> its

File: src/main/java/org/terasology/gooeyDefence/EnemyManager.java [hunk 3]
```diff
@@ -93,18 +72,24 @@ public void onFieldActivated(OnFieldActivated event, EntityRef entity) {
...
         enemies.stream().filter(enemy -> enemy.hasComponent(EntrancePathComponent.class))
-                .forEach(enemy -> enemy.getComponent(EntrancePathComponent.class).setPathManager(pathfindingSystem));
+                .forEach(enemy -> enemy.getComponent(EntrancePathComponent.class).setPathManager(pathfindingManager));
 
         //delayManager.addPeriodicAction(DefenceField.getShrineEntity(), "SpawnEnemyEvent", 500, 500);
     }
 
+    /**
+     * Test handler to allow easy enemy spawning
+     */
```
Comment (vampcat) [inline src/main/java/org/terasology/gooeyDefence/EnemyManager.java]: Either use normal comments (//) or add javadocs for params and return type.

File: src/main/java/org/terasology/gooeyDefence/EnemyManager.java [hunk 4]
```diff
@@ -93,18 +72,24 @@ public void onFieldActivated(OnFieldActivated event, EntityRef entity) {
...
     @ReceiveEvent
     public void onActivate(ActivateEvent event, EntityRef entity) {
         for (int i = 0; i < DefenceField.entranceCount(); i++) {
             spawnEnemy(i);
         }
     }
 
+    /**
+     * Test handler to allow easy enemy spawning
```
Comment (vampcat) [inline src/main/java/org/terasology/gooeyDefence/EnemyManager.java]: Same as above ^^

File: src/main/java/org/terasology/gooeyDefence/components/towers/TowerComponent.java [hunk 5]
```diff
@@ -21,11 +21,12 @@
 import java.util.Set;
 
 /**
- * Component storing the ID's of all the blocks that make up a tower.
+ * Component for the abstract tower entity.
+ * Stores the ID's of all the blocks that make up the tower.
```
Comment (vampcat) [inline src/main/java/org/terasology/gooeyDefence/components/towers/TowerComponent.java]: IDs, not ID's.

File: src/main/java/org/terasology/gooeyDefence/events/combat/DoSelectEnemies.java [hunk 6]
```diff
@@ -17,22 +17,40 @@
...
 import java.util.HashSet;
 import java.util.Set;
 
+/**
+ * Event sent to select the enemies that will be attacked.
+ * Sent against the Emitter blocks in the tower.
+ *
+ * @see TowerTargeter
+ */
 public class DoSelectEnemies implements Event {
```
Comment (vampcat) [inline src/main/java/org/terasology/gooeyDefence/events/combat/DoSelectEnemies.java:32]: Rename DoSelectEvents to SelectEnemeiesEvent?

File: src/main/java/org/terasology/gooeyDefence/towerBlocks/base/TowerTargeter.java [hunk 7]
```diff
@@ -15,7 +15,16 @@
  */
 package org.terasology.gooeyDefence.towerBlocks.base;
 
-public abstract class TowerEmitter {
+/**
+ * Base class for all the Targeter blocks.
+ * <p>
+ * Targeters select the enemies the tower will attack.
+ * They require power, provided by {@link TowerCore}'s
```
Comment (vampcat) [inline src/main/java/org/terasology/gooeyDefence/towerBlocks/base/TowerTargeter.java]: TowerCores, not TowerCore's.

File: src/main/java/org/terasology/gooeyDefence/towerBlocks/targeters/SingleTargetTargeterComponent.java [hunk 8]
```diff
@@ -13,10 +13,16 @@
...
-package org.terasology.gooeyDefence.towerBlocks.emitters;
+package org.terasology.gooeyDefence.towerBlocks.targeters;
 
 import org.terasology.entitySystem.Component;
-import org.terasology.gooeyDefence.towerBlocks.base.TowerEmitter;
+import org.terasology.gooeyDefence.towerBlocks.base.TowerTargeter;
 
-public class SingleTargetEmitterComponent extends TowerEmitter implements Component {
+/**
+ * Targets the first enemy in it's range.
```
Comment (vampcat) [inline src/main/java/org/terasology/gooeyDefence/towerBlocks/targeters/SingleTargetTargeterComponent.java]: It's -> its

**Review by vampcat — CHANGES_REQUESTED** [review]
My biggest gripe : Effector. It should actually be Affector, since it affects the attack type.

File: assets/prefabs/blocks/effectors/EffectorOne.prefab (hunk 1 above)
Comment (syntaxi) [reply assets/prefabs/blocks/effectors/EffectorOne.prefab:6]: File is actually deleted in later branches so this is "fixed"

File: src/main/java/org/terasology/gooeyDefence/events/combat/DoSelectEnemies.java (hunk 6 above)
Comment (syntaxi) [reply src/main/java/org/terasology/gooeyDefence/events/combat/DoSelectEnemies.java:32]: (Renamed in a later branch to avoid conflicts)

**Pull request merged**